### PR TITLE
Port git dit list

### LINKED ISSUE #41: Port "git-dit-list" to rust
What the title says. Porcelain commands should go into the binary.

### PR DESCRIPTION
**Pull request by neithernut**
Resolves #41.

### CODE REVIEW

**Comment by neithernut** [comment]
I'm not yet sure what to do about time formatting. I guess we need/want an external crate, like [chrono](https://github.com/chronotope/chrono). Any preferences/suggestions/ideas @matthiasbeyer?

**Comment by matthiasbeyer** [comment]
chrono it is.

File: src/main.rs [hunk 1]
```diff
@@ -144,6 +146,52 @@ fn get_issue_tree_init_hashes(repo: &Repository, _: &clap::ArgMatches) -> i32 {
 
 // Porcelain subcommand implementations
 
+/// list subcommand implementation
+///
+fn list_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
+    // get initial commits
+    let mut commits : Vec<Commit>= try_or_1!(repo.get_all_issue_hashes())
```
Comment (matthiasbeyer) [inline src/main.rs]: `s/>=/> =/` ?

File: src/main.rs [hunk 2]
```diff
@@ -144,6 +146,52 @@ fn get_issue_tree_init_hashes(repo: &Repository, _: &clap::ArgMatches) -> i32 {
...
+        .abort_on_err()
+        .map(|oid| repo.find_commit(oid))
+        .abort_on_err()
+        .collect();
+
+    // descending order, maybe limited to some number specified by the user
+    commits.sort_by(|a, b| b.time().cmp(&a.time()));
```
Comment (matthiasbeyer) [inline src/main.rs]: I'm rather sure we can do this within the iteration above (before the `collect()` call) using the `itertools` crate.

**Comment by matthiasbeyer** [comment]
Dang. I was too fast, wanted to block the PR. Sorry. Next time I'll do better!

File: src/main.rs (hunk 2 above)
Comment (neithernut) [inline src/main.rs]: Sorting an array should be more efficient.

Comment (matthiasbeyer) [inline src/main.rs]: Sorting 1,000,000 `i64`'s:

* without itertools: 1.713s
* with itertools: 1.718s

So I guess... no! :smile: 

But I don't care.

**Pull request merged**